src/validation.cairo

```diff
@@ -350,4 +386,182 @@ mod tests {
+                            previous_output: OutPoint {
+                                txid: 0_u256, vout: 0x1_u32, txo_index: 0,
+                            },
+                            witness: array![].span()
+                        }
+                    ]
+                        .span(),
+                    outputs: array![
+                        TxOut {
+                            value: 5000000000_u64,
+                            pk_script: @from_base16(
+                                @"4104678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38c4f35504e51ec112de5c384df7ba0b8d578a4c702b6bf11d5fac"
+                            ),
+                        }
+                    ]
+                        .span(),
+                    lock_time: 0
+                }
+            ]
+                .span()
+        };
+        let total_fees = 5000000000_u64;
+        let block_height = 1;
+
+        validate_coinbase(@block, total_fees.into(), block_height);
+    }
+
+    #[test]
+    #[should_panic(expected: ('txid should be 0',))]
+    fn test_validate_coinbase_with_txid_not_zero() {
+        let block = Block {
+            header: Header { version: 1_u32, time: 1231006505_u32, bits: 1, nonce: 2083236893_u32 },
+            txs: array![
+                Transaction {
+                    version: 1,
+                    is_segwit: false,
+                    inputs: array![
+                        TxIn {
+                            script: @from_base16(
+                                @"01091d8d76a82122082246acbb6cc51c839d9012ddaca46048de07ca8eec221518200241cdb85fab4815c6c624d6e932774f3fdf5fa2a1d3a1614951afb83269e1454e2002443047"
+                            ),
+                            sequence: 4294967295,
+                            previous_output: OutPoint {
+                                txid: 0x2_u256, vout: 0xFFFFFFFF_u32, txo_index: 0,
+                            },
+                            witness: array![].span()
+                        }
+                    ]
+                        .span(),
+                    outputs: array![
+                        TxOut {
+                            value: 5000000000_u64,
+                            pk_script: @from_base16(
+                                @"4104678afdb0fe5548271967f1a67130b7105cd6a828e03909a67962e0ea1f61deb649f6bc3f4cef38c4f35504e51ec112de5c384df7ba0b8d578a4c702b6bf11d5fac"
+                            ),
+                        }
+                    ]
+                        .span(),
+                    lock_time: 0
+                }
+            ]
+                .span()
+        };
+        let total_fees = 5000000000_u64;
+        let block_height = 1;
+
+        validate_coinbase(@block, total_fees.into(), block_height);
+    }
+    // #[test]
+// #[should_panic(expected: ('total output > block rwd + fees',))]
+// fn test_validate_coinbase_outputs_amount() {
```

Comment:
I've been trying to make this test pass but I'm not sure what I'm missing @maciejka @m-kus,

Reply:
There are compilation errors:
![image](https://github.com/user-attachments/assets/a3a8b4c1-ab7b-449a-bb48-7f8fa810543a)
you need to mock witness with `@""` to fix this one.

![image](https://github.com/user-attachments/assets/a6590bad-bf9a-4b4d-8134-4e40d56776a3)
add `?` at the end of the validate_coinbase function

Reply:
I'm not getting the error on "witness" locally, all the other 3 tests are passing except the last one (test_validate_coinbase_outputs_amount)
<img width="854" alt="Screenshot 2024-08-13 at 9 38 43 AM" src="https://github.com/user-attachments/assets/521d3f30-7ac6-4006-bfd0-f713ddccee2a">

Reply:
did you push your recent changes?

Reply:
That screen shot was from before I made the requested changes.